### PR TITLE
Export leading icon types

### PR DESCRIPTION
**Pull request by ddouglasz**
#### Summary

Export leading icon types so it can be properly consumed the right way. 

For more context, see https://github.com/commercetools/merchant-center-frontend/pull/16448/files#r1525033953

### CODE REVIEW

**Comment by CarlosCortizasCT** [comment]
Please @ddouglasz, take a look at the alternative proposal I'm suggesting in [this PR](https://github.com/commercetools/ui-kit/pull/2753) 🙏

**Comment by ddouglasz** [comment]
> Please @ddouglasz, take a look at the alternative proposal I'm suggesting in [this PR](https://github.com/commercetools/ui-kit/pull/2753) 🙏

Yes the change proposal is clear, it makes sense to export the `TLeadingIconProps` from it's own icon-types file in the root of the `leading-icon` folder so that we can import both the `leading-icon` component and it's type from the same directory.  
Thank you.

File: .changeset/mean-crabs-perform.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-uikit/icons': minor
+---
+
+export types for leading icon
```
Comment (CarlosCortizasCT) [inline .changeset/mean-crabs-perform.md:5]:
```suggestion
We now exports props typescript types for both `LeadingIcon` and `InlineSvg` components.
```

Comment (ddouglasz) [reply .changeset/mean-crabs-perform.md]: updated here: https://github.com/commercetools/ui-kit/pull/2752/commits/e96762b7c0e01ecdbbbe20671cb4d3a1ba5ef25a

**Pull request merged**